### ISSUE TITLE
Support built in taichi kernels in AOT 

### ISSUE THREAD

**Issue by ailzhang** [taichi-dev/taichi#4585]
related: https://github.com/taichi-dev/taichi/issues/3642
When users exchange data between taichi and python, e.g. when they write `field.fill(0)` in python, a built-in taichi kernel is compiled and executed. https://github.com/taichi-dev/taichi/blob/master/python/taichi/_kernels.py#L16 
A problem for AOT users is that these kernels are "invisible" to them so that they don't usually add them into their aot module. IMHO there are two ways to fix it:
1. Users can copy-paste these built-in kernels into their script and explicitly add them in the aot module. (I'm currently using this approach in the demo)
2. We should provide similar interfaces in C++, e.g `ndarray.copy_from, ndarray.fill` etc in https://github.com/taichi-dev/taichi/blob/master/taichi/aot/module_loader.h#L22 so that we guarantee a similar experience compared to python frontend.  Note in this approach we'll need to record the built-in kernels used and add them into the aot module automatically. It might require a bit tweak as we use templated kernel for them. 

IMHO the second solution is the long term solution that we want to pursue, suggestions / discussions are welcome! 
cc: @k-ye  @ghuau-innopeak @qiao-bo @bobcao3

**Comment by k-ye** [comment]
I agree that the 2nd approach is better. Our compute graph system should also be able to identify these builtin APIs.

**Comment by ghuau-innopeak** [comment]
@ailzhang Same here, the 2nd approach is much more flexible and easier on the long term, especially with a transparent interface where the user won't have to explicitly add/load them to/from the AOT module.
My only concern would be when the built-in API will have hundred of functions being exported and loaded but I don't believe this is the case right now :), in this case, we might want to allow enabling/disabling exporting the core `built-in` API, or even filter them.
